solr/core/src/java/org/apache/solr/servlet/CoordinatorHttpSolrCall.java

```diff
@@ -208,9 +208,12 @@ private static void setMDCLoggingContext(String collectionName) {
   private static void addReplica(String syntheticCollectionName, CoreContainer cores) {
     SolrQueryResponse rsp = new SolrQueryResponse();
     try {
+      String coreName = syntheticCollectionName + "_" + "r1";
```

Comment:
Do we want to include node name as a part of the core name? otherwise 2 coordinator nodes might use same name for the replica?

Reply:
I thought about that. Core names do not have to be unique

Reply:
Good to know! 👍🏼 

In this case we might want to always call `waitForState` as in https://github.com/apache/solr/pull/1762/files#diff-eedf409265fc219f98f193ae89d3f1b09df78fe49f70bb5b9eaa6c6ff46e6ac7R143 since `addReplica` can now return while the target replica is still under construction by another request.

Reply:
Ah I think we do need to add node name, as the [AddReplicaCmd logic](https://github.com/apache/solr/blob/main/solr/core/src/java/org/apache/solr/cloud/api/collections/AddReplicaCmd.java#L388) would iterate through all slices and throw exception on any core name matches.

Therefore, for the 1st coordinator node, it's fine -> the replica is created by the collection creation call, which follows the standard name, ie `.sys.COORDINATOR-COLL-conf_shard1_replica_n1`.

The 2nd coordinator node is also fine -> `.sys.COORDINATOR-COLL-conf_r1`.

However starting from the 3rd coordinator, it would fail with infinite loop, since it will skip adding replica (due to the check linked above), but `solrCall.getCoreByCollection(syntheticCollectionName, isPreferLeader)` would not be able to load such core (since it's not in 3rd node), hence causing the infinite loop.

Such can be reproduced by modifying node count [here](https://github.com/apache/solr/pull/1762/files#diff-13f37ffd1807b53dacbf21317f6f6d3114448ae2e85c90835fa2f368e31b3cbaR489) in the test case from 2 to 3

Reply:
true. I'm not even sure why someone added that check

Reply:
Thank you @noblepaul !

I added another small commit 45654e1c5bff66413956202cade3d1b75733be17 to ensure that the fix work:
1. Changed coordinator node count 2 -> 4
2. Verified the replica count on the synthetic collection

Also a minor change to the addReplica flow, that we always check for replica status afterwards (since `addReplica` might now return if exception is thrown and caught, and with replica status not always active, we could run into infinite call loop, this is a rather rare case, but it doesn't hurt to check.